tests/upgrades/test_repository.py

```diff
@@ -0,0 +1,87 @@
+"""Test for Repository related Upgrade Scenario's
+
+:Requirement: Upgraded Satellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fabric.api import run
+from nailgun import entities
+from robottelo.test import APITestCase
+from robottelo.api.utils import create_sync_custom_repo
+from upgrade_tests import post_upgrade, pre_upgrade
+from upgrade_tests.helpers.scenarios import create_dict, get_entity_data
+
+
+class Scenario_repository_upstream_authorization_check_Bug1641785(APITestCase):
```

Comment:
I'd prefer to record the associated bugs in docstring rather than the title, at least that's the case in most of robottelo, for example like this
https://github.com/SatelliteQE/robottelo/blob/2e8b05292e3f017c8a10bf4375890517ebc9384f/tests/foreman/ui/test_contenthost.py#L122
It is more flexible, you can add more bugs if they appear, whereas class name changes tend to cause trouble in result analysis

Reply:
@pondrejk yeah that make sense . thanks 
that is updated the same now.